packages/gatsby/src/utils/flags.ts

```diff
@@ -60,6 +60,15 @@ const activeFlags: Array<IFlag> = [
     description: `Don't process images during development until they're requested from the browser. Speeds starting the develop server.`,
     umbrellaIssue: `https://github.com/gatsbyjs/gatsby/discussions/27603`,
   },
+  {
+    name: `FAST_REFRESH`,
+    env: `GATSBY_FAST_REFRESH`,
+    command: `develop`,
+    telemetryId: `FastRefresh`,
+    experimental: false,
+    description: `Replaces the default HMR loader (react-hot-loader) with React Fast Refresh. Recommended for versions of React >= 17.0.`,
```

Comment:
I can't really come up with something better but `HMR` acronym which is well understood for maintainers, might be not clear for users - maybe we can drop this all together and just mentions that this will use "Fast Refresh" and not legacy "react-hot-loader"?